### PR TITLE
fixes regarding down

### LINKED ISSUE #68: Stop independent services in kube-compose defined in docker-compose.yml
kube-compose down should be able to stop independent services.

For eg:  kube-compose down service-1

### LINKED ISSUE #75: Validate the names of services passed to up or down like docker-compose does

### PR DESCRIPTION
**Pull request by jbrekelmans**
1. fixes #75: validate names of service names passed to up and down commands
2. fixes #68: down for specific services
3. move code that needs to be used in `down.go` from `up.go` to a common package
4. simplify `up.go` by adding a field for the docker compose service to the app struct, rather than performing many map lookups
5. move `linux.go` to linux internal package
6. move `userinfo.go` to docker package
7. move docker package to into internal
8. increase coverage in various places

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -31,8 +31,8 @@ func init() {
 	rootCmd.PersistentFlags().StringVarP(namespace, "namespace", "n", "", "namespace for environment")
 	rootCmd.PersistentFlags().StringP("file", "f", "", "Specify an alternate compose file")
 	envID := new(string)
-	rootCmd.PersistentFlags().StringVarP(envID, "env-id", "e", "", "used to isolate environments deployed to a shared namespace, "+
-		"by (1) using this value as a suffix of pod and service names and (2) using this value to isolate selectors(required)")
+	rootCmd.PersistentFlags().StringVarP(envID, "env-id", "e", "", "(required) used to isolate environments deployed to a shared namespace, "+
```
Comment (xUnholy) [inline cmd/root.go]: Should set this flag to be required

`rootCmd.PersistentFlags().StringVarP(envID, "env-id", "e", "", "(required) used to isolate environments deployed to a shared namespace, "+
		"by (1) using this value as a suffix of pod and service names and (2) using this value to isolate selectors(required)")			"by (1) using this value as a suffix of pod and service names and (2) using this value to isolate selectors")
rootCmd. MarkPersistentFlagRequired("env-id")`

If it's set but with no value then check ENV vars?

Comment (jbrekelmans) [reply cmd/root.go]: I just changed the message, anyway does setting it to required make cobra fail if there is no flag but there is an environment variable?

Comment (xUnholy) [reply cmd/root.go]: fails if the flag is not set, I think if it's required the flag should be set, then we validate either the input or the ENV var.

Comment (xUnholy) [reply cmd/root.go]: Otherwise, it's confusing saying it's required but we don't set it required until after we check its non-empty or no ENV var??

**Comment by xUnholy** [comment]
@jbrekelmans need to fix merge conflicts

File: cmd/root.go (hunk 1 above)
Comment (jbrekelmans) [reply cmd/root.go]: I just looked at the code, and looks like the flag is made required only if the environment variable is not set.

**Comment by jbrekelmans** [comment]
I only have a Windows laptop at the moment, and this should probably be tested before merging.

File: cmd/common.go [hunk 2]
```diff
@@ -56,9 +56,13 @@ func upOrDownCommandCommon(cmd *cobra.Command, args []string) (*config.Config, e
 	if namespace, _ := cmd.Flags().GetString("namespace"); namespace != "" {
 		cfg.Namespace = namespace
 	}
-	cfg.Services = map[string]bool{}
-	for _, arg := range args {
-		cfg.Services[arg] = true
+	if len(args) == 0 {
```
Comment (xUnholy) [inline cmd/common.go]: `if len(args) == 0 {` >> `if len == ""` ?

File: internal/pkg/docker/userinfo_test.go [hunk 3]
```diff
@@ -1,29 +1,16 @@
-package up
+package docker
```
Comment (xUnholy) [inline internal/pkg/docker/userinfo_test.go]: It's also a good idea we look at testing with importing the package. Perhaps a TODO for a later discussion around testing

File: cmd/common.go (hunk 2 above)
Comment (jbrekelmans) [reply cmd/common.go]: It's a slice not a string

File: internal/pkg/docker/userinfo_test.go (hunk 3 above)
Comment (jbrekelmans) [reply internal/pkg/docker/userinfo_test.go]: What do you mean?

Comment (xUnholy) [reply internal/pkg/docker/userinfo_test.go]: https://segment.com/blog/5-advanced-testing-techniques-in-go/

Comment (xUnholy) [reply internal/pkg/docker/userinfo_test.go]: > Use a separate _test package

Comment (jbrekelmans) [reply internal/pkg/docker/userinfo_test.go]: How do we test internal functions (lowercase first letter) if we should use a separate `_test` package?

**Pull request merged**